### PR TITLE
Add evm_block_height metric

### PR DESCRIPTION
**Pull request by illia-malachyn**
This covers "EVM height should be submitted as a value on event ingestion"

## Summary by CodeRabbit

- **New Features**
	- Enhanced event ingestion process with a new metrics collector integration.
	- Introduced methods for tracking EVM block heights in the metrics collection.
	- Improved metrics reporting capabilities in the event ingestion engine.

- **Bug Fixes**
	- Enhanced accuracy of event processing metrics reporting.

- **Documentation**
	- Updated documentation to reflect new methods and metrics integration.

- **Tests**
	- Added unit tests for the new metrics functionalities.

### CODE REVIEW

File: metrics/collector.go [hunk 1]
```diff
@@ -24,6 +26,11 @@ func NewCollector(logger zerolog.Logger) Collector {
 		Help: "Total number of errors returned by the endpoint resolvers",
 	})
 
+	evmBlockHeight := prometheus.NewGauge(prometheus.GaugeOpts{
```
Comment (illia-malachyn) [inline metrics/collector.go:29]: TODO: should use counter instead of gauge.

File: services/ingestion/engine.go [hunk 2]
```diff
@@ -151,6 +155,7 @@ func (e *Engine) processEvents(events *models.CadenceEvents) error {
 	if err != nil {
 		return fmt.Errorf("failed to index block %d event: %w", events.Block().Height, err)
 	}
+	e.collector.EvmBlockIndexed()
```
Comment (sideninja) [inline services/ingestion/engine.go:158]: this should go at the end of the function, and it should take in the EVM height which was indexed

File: metrics/collector.go [hunk 3]
```diff
@@ -58,6 +66,10 @@ func (c *DefaultCollector) ApiErrorOccurred() {
 	c.apiErrorsCounter.Inc()
 }
 
+func (c *DefaultCollector) EvmBlockIndexed() {
+	c.evmBlockHeight.Inc()
```
Comment (sideninja) [inline metrics/collector.go:70]: we should take in the height as a value, there might be cases where we will re-index already indexed values (crash recovery)

**Review by sideninja — COMMENTED** [review]
left a comment on height

**Review by sideninja — APPROVED** [review]

**Pull request merged**